hw1/task1/averaging.py

```diff
@@ -0,0 +1,48 @@
+import pandas as pd
+import numpy as np
+
+
+def average_gen(general_stats):
+    avg_data = pd.DataFrame(columns=['mean_cost',
+                                     'longest_ride',
+                                     'max_count',
+                                     'max_count_start',
+                                     'max_count_end',
+                                     'invalid_rows'])
+    avg_data['mean_cost'] = [(general_stats['mean_cost'] * general_stats['count'] / general_stats['count'].sum()).sum()]
+    avg_data['longest_ride'] = [general_stats['longest_ride'].max()]
+    max_count_index = np.where(general_stats['max_count'] == general_stats['max_count'].max())[0][0]
+    avg_data['max_count'] = general_stats['max_count'].iloc[max_count_index]
+    avg_data['max_count_start'] = general_stats['max_count_start'].iloc[max_count_index]
+    avg_data['max_count_end'] = general_stats['max_count_end'].iloc[max_count_index]
+    avg_data['invalid_rows'] = general_stats['invalid_rows'].sum()
+
+    return avg_data
+
+
+def average_missing(missing_stats):
+    misses = pd.DataFrame(columns=['missing_dates'])
+    misses['missing_dates'] = missing_stats['missing_dates'].unique()
+    return misses
```

Comment:
we're interested in the dates, that are missing between min and max dates of all the input files.
I.e. if in one file we have dates for 2019-01 and in the other for 2020-01, we'll get all the dates from 2019-02 through 2019-12